corehq/apps/hqwebapp/templatetags/hq_shared_tags.py

```diff
@@ -103,6 +104,14 @@ def concat(str1, str2):
     resource_versions = {}
 
 
+@register.filter
+def pp_json(data):
```

Comment:
This behaves differently from the `JSON` filter in an important way. `pp_json` will try to load string values and dump them as JSON in order to pretty-print JSON strings, whereas `JSON` will simply dump the string.